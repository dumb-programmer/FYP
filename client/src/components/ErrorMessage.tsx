export default function ErrorMessage({ message }: { message?: string }) {
    return <span className="text-sm text-red-600">{message}</span>;
}