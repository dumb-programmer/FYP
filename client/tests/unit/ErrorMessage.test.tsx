import React from "react";
import { render, screen } from "@testing-library/react";
import { expect, describe, it } from "vitest";
import ErrorMessage from "../../src/components/ErrorMessage";

const renderComponent = () => render(<ErrorMessage message="something went wrong" />);

describe("ErrorMessage", () => {
    it("renders correctly", () => {
        renderComponent();

        const errorMessage = screen.getByText(/something went wrong/);

        expect(errorMessage).toMatchSnapshot();
    });
})