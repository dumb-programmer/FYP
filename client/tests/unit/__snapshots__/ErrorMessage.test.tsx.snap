// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorMessage > renders correctly 1`] = `
<span
  class="text-sm text-red-600"
>
  something went wrong
</span>
`;
